### ISSUE TITLE
[Request] Split doctest source files

### ISSUE THREAD

**Issue by TheQwertiest** [onqtam/doctest#330]
Currently doctest (before concatenation) consists of two files:
`doctest_fwd.h`: 2604 LOC
`doctest.cpp`: 3330 LOC

These files are HUGE and very hard to parse and understand for contributors unfamiliar with `doctest` codebase.

Splitting these files (into logical units) will lower the time required for new contributors to actually start contributing. And since all source files are concatenated anyway, it won't affect anything (aside from having to modify `cmake` files a bit).

**Comment by onqtam** [comment]
That's true and indeed necessary - I've been putting it off for ages and I guess I won't be able to get to doing this in the near future either. But I'm keeping this issue open until this is done.

**Comment by mitchgrout** [comment]
Closed in favour of #941 (also, fully complete now!)
